tests/unit_tests/encoders/text/test_short_text_encoder.py

```diff
@@ -20,27 +20,38 @@ def test_get_tokens(self):
         assert tokenize_text("don't wouldn't") == ['do', 'not', 'would', 'not']
 
     def test_concat(self):
-        self._test_concat(onehot=True)
-        self._test_concat(onehot=False)
+        self._test_concat(True, True)
+        self._test_concat(False, False)
+        self._test_concat(True, False)
+        self._test_concat(False, True)
     
     def test_mean(self):
-        self._test_mean(onehot=True)
-        self._test_mean(onehot=False)
+        self._test_mean(True, True)
+        self._test_mean(False, False)
+        self._test_mean(True, False)
+        self._test_mean(False, True)
 
-    def _test_concat(self, onehot):
+    def _test_concat(self, onehot, is_target):
         vocab_size = 99 if onehot else 800
         
         priming_data = generate_sentences(2, 6, vocab_size)
         test_data = random.sample(priming_data, len(priming_data) // 5)
 
-        enc = ShortTextEncoder(combine='concat')
+        enc = ShortTextEncoder(is_target=is_target)
         enc.prepare_encoder(priming_data)
 
         if onehot:
```

Comment:
I think at this point, with 4 possible states for this function, it might be better to make separate tests. Yes, this will make the code of tests repetitive, but it will be much more readable. It will also make it easier to debug when a test fails: if there is an exception, you will not only get the info that `test_mean` failed, but you will get the info that `test_info_onehot_not_target` failed.